surfman/src/connection.rs

```diff
@@ -55,9 +55,15 @@ pub trait Connection: Sized {
     ) -> Result<Self::Device, Error>;
 
     /// Opens the display connection corresponding to the given raw display handle.
-    #[cfg(feature = "sm-raw-window-handle")]
+    #[cfg(feature = "sm-raw-window-handle-05")]
+    fn from_raw_display_handle_05(
+        raw_handle: rwh_05::RawDisplayHandle,
+    ) -> Result<Self, Error>;
```

Comment:
Instead of using the deprecated `RawWindowHandle` and `RawDisplayHandle` when `raw-window-handle` 0.6 is enabled, let's switch to using `DisplayHandle` and `WindowHandle`. That way instead of pairs of functions like:

 - `from_raw_display_handle_05`
 - `from_raw_display_handle`

We could have:

 - `from_display_handle`
 - `from_raw_display_handle`

Reply:
With rwh 0.6, the non-raw API's central idea is ensuring that whatever entity (e.g. a Windows Window) backs the raw window handle (e.g. HWND) isn't destroyed until the reference to the handle is dropped.

Should I be using something like: `has_display_handle: Box<dyn HasDisplayHandle + 'static>` or `struct WidgetThatsCreatedFromDisplayHandle<'a>{ display_handle: DisplayHandle<'a> }` ?

I'm not familiar with GATs, but we might even be able to make each of the associated types on the Connection trait generic over the HasDisplayHandle and HasWindowHandle implementations?

Reply:
This seems like a larger refactor. For now can we simply create and consume these handles in the moment and then figure out how to keep the display primitives alive in a separate change?

Reply:
Alright, that should do it.
- No more _05 suffixes
- sm-raw-w-indow-handle-06 feature exposes an interface that consumes DisplayHandles and returns surfman objects that contain their raw platform-specific handle.
Thanks for your help!